### PR TITLE
Bug 1390173 - Test more pod to pod connectivity test combinations

### PR DESCRIPTION
**Pull request by pravisankar**
Created 2 isolated and 2 global namespaces on every node, so that pod to pod connectivity
tests can be performed among same/different namespaces, same/different nodes and global/non-global pods.

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/networking PTAL

**Comment by knobunc** [comment]
[test]

File: pkg/diagnostics/network/run_pod.go [hunk 1]
```diff
@@ -74,7 +76,7 @@ func (d *NetworkDiagnostic) Check() types.DiagnosticResult {
 		return d.res
 	}
 	if !ok {
-		d.res.Warn("DNet2002", nil, fmt.Sprintf("Skipping network diagnostics check. Reason: Not using openshift network plugin."))
+		d.res.Warn("DNet2002", nil, "Skipping network diagnostics check. Reason: Not using openshift network plugin.")
```
Comment (danwinship) [inline pkg/diagnostics/network/run_pod.go]: you should split out cleanups like this into separate commits

File: pkg/diagnostics/networkpod/pod.go [hunk 2]
```diff
@@ -148,6 +148,13 @@ func (d CheckPodNetwork) checkConnection(pods1, pods2 []kapi.Pod, warnMsg string
 			}
 		}
 	}
+
+	if !sameNamespace && !strings.Contains(warnMsg, "global to non-global projects") {
```
Comment (danwinship) [inline pkg/diagnostics/networkpod/pod.go]: Why don't you warn in that case? Is there some better way of testing for that than doing a substring check on the warning message?

Comment (pravisankar) [inline pkg/diagnostics/networkpod/pod.go]: We don't want to warn in this case because same namespace can't have global and non-global pods at the same time. May be we can make the caller pass a flag to checkConnection() to check this condition or not.

**Comment by pravisankar** [comment]
Separated bug fix and cleanup. 
Caller of checkConnection() will tell whether to check same namespace condition.
@knobunc @danwinship updated, ptal

**Comment by danwinship** [comment]
LGTM
flake is #9076, [test]

**Comment by danwinship** [comment]
flake is #10773 
[test]

**Comment by knobunc** [comment]
[merge]
